### PR TITLE
Automate deployment using podman and systemd

### PR DESCRIPTION
**Pull request by morgaesis**
Use systemd to schedule daily update checks and rebuild bot container if any updates are on git master branch.

### CODE REVIEW

File: the-lone-dancer.service [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=The Lone Dancer
+After=basic.target
+
+[Service]
+Type=simple
+#Restart=always
+#RemainAfterExit=yes
+User=root
```
Comment (michael-je) [inline the-lone-dancer.service:9]: Is this necessary?

**Comment by michael-je** [comment]
Is the `lone-dancer.service` necessary? :o

**Comment by morgaesis** [comment]
Woops, no not needed at all. It's just a copy of a service I had on my system.

**Comment by morgaesis** [comment]
Later this should probably be changed to have proper dependencies on each distro and running on bare metal rather than in container.

File: the-lone-dancer.service (hunk 1 above)
Comment (michael-je) [reply the-lone-dancer.service:9]: Oh sorry, I was referring to line number 9. Is it necessary to run the script as root?

Comment (michael-je) [reply the-lone-dancer.service:9]: -

Comment (morgaesis) [reply the-lone-dancer.service:9]: No, it does not need to be run as root, but it simplifies deployment.

This was a quick-'n-dirty setup. I need to create a new user, though for running the container.

I thought it was standard to run containers as root, but [best practices](https://cloud.google.com/architecture/best-practices-for-operating-containers) says we shouldn't.

Do not merge until this is resolved!

**Comment by morgaesis** [comment]
Opted to still run in a container, but as `the-lone-dancer` user. This still gives auto-updates right from GitHub. Future releases should have bare-metal installation routes as well.

**Comment by morgaesis** [comment]
Still need reviews on this @thorduragust @gardarandri

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#/bin/bash
```
Comment (gardarandri) [inline setup.sh:1]: Please add a comment explaining what this script does and an example invocation.

File: teardown.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#/bin/bash
```
Comment (gardarandri) [inline teardown.sh:1]: Please add a comment explaining what this script does and an example invocation.

File: update.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (gardarandri) [inline update.sh:1]: Please add a comment explaining what this script does and an example invocation.

**Review by gardarandri — APPROVED** [review]

File: setup.sh (hunk 2 above)
Comment (morgaesis) [reply setup.sh:1]: Done.

File: teardown.sh (hunk 3 above)
Comment (morgaesis) [reply teardown.sh:1]: Done.

**Comment by morgaesis** [comment]
Sorry this is taking long.

How do you feel about making the update as `root`? I need to be able to start/stop the main service, but this can only be done with elevated privileges. We could add `the-lone-dancer` to the `sudoers` file allowing sudo on start/stop of the service, but I don't like automating messing with the `sudoers` file, and don't want the installation to take manual steps.

@gardarandri @michael-je

**Comment by michael-je** [comment]
Are you sure it isn't possible to update without starting/stopping the service? I really don't like the idea of random programs/scripts asking for superuser priveledges unless strictly required for the program to function. 

If you're absolutely sure, then even though I don't like it, I guess it's fine as long as the user recieves a clear notice during the installation process and a statement that installing it via systemd is **optional**.

Editing the sudoers file is a definite no-go for me.

**Comment by morgaesis** [comment]
We could install on bare-metal with the user `the-lone-dancer`, but then there is no built-in auto-update. To auto-update you'd have to set it up yourself, or we compile system packages for the various distros and distribute either through the distro package distribution or by making the system package manager fetch updates from this repo.

**Comment by michael-je** [comment]
I see, Yeah you're right that sounds way out of scope for the project haha. Let's go with this solution then :)

Just out of curiosity, what would the user have to do to set up the autoupdates themselves?

**Comment by morgaesis** [comment]
To auto-update: cron-job, systemd timers. Both would still require sudo privileges to mess with configuration and restart the bot (as the `the-lone-dancer` user). This is what system updates would do.

To be sure, which route should we go?
- [ ] Containerized, auto-updating, `root`-updater
- [ ] Bare metal, `root` install, no updates
- [ ] Anything not requiring `root`, auto-updates
- [ ] Anything not requiring `root`, no auto-updates

**Pull request merged**